SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/TreeFactory.swift

```diff
@@ -35,10 +36,17 @@ class TreeFactory {
 		the root node having 'rows' chlidnodes, where each of these nodes
 		have 'cols' buttonNodes as children
 	*/
-	class func treeForRowColumnScanning (rows: Int, cols: Int) -> Tree {
+    //
+    //dummy_num is the number of bummy button in this tree
+
+    class func treeForRowColumnScanning (rows: Int, cols: Int, dummy_num: Int) -> Tree {
 		let T = Tree()
 		T.treeType = .ROW_COLUMN
 		T.size = rows * cols
+       // compute the position of dummy buttons according to the number of dummy button
+        let ex_col = cols - dummy_num % cols
```

Comment:
It isn't really clear to me why these variables are called `ex_row` and `ex_col`, even after reading through the code. Please change these names to something more descriptive that is also in `camelCase`.